internal/plugin/connectors/tcp/pg/startup.go

```diff
@@ -11,24 +11,23 @@ import (
 func (s *SingleUseConnector) Startup() error {
 	s.logger.Debugf("Handling connection %+v -> %+v", s.clientConn.RemoteAddr(), s.clientConn.LocalAddr())
 
-	messageBytes, err := protocol.ReadStartupMessage(s.clientConn)
+	version, options, err := s.parseStartupMessage()
 	if err != nil {
 		return err
 	}
 
-	version, options, err := protocol.ParseStartupMessage(messageBytes)
-	if err != nil {
-		return err
-	}
-
-	s.logger.Debugf(
-		"s.Client version: %v, (SSL mode: %v)",
-		version,
-		version == protocol.SSLRequestCode)
-
 	// Handle the case where the startup message was an SSL request.
 	if version == protocol.SSLRequestCode {
-		return fmt.Errorf("SSL not supported")
+		s.sslNotSupported()
+		version, options, err = s.parseStartupMessage()
+		if err != nil {
+			return err
+		}
+
+		// There should not be a second SSLRequest
+		if version == protocol.SSLRequestCode {
+			return fmt.Errorf("SSL Request after SSL not supported response")
```

Comment:
Sorry after some reflection I think this is a better message
```suggestion
			return fmt.Errorf("Unexpected SSL Request after SSL not supported response")
```